minecraftServers/Controllers/ServersController.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using minecraftServers.Repositories;
+using minecraftServers.Servers;
+using System.Text.RegularExpressions;
+
+namespace minecraftServers.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class ServersController : ControllerBase
+{
+    private readonly IServersRepository _serversRep;
+    private readonly List<ServerList> _servers = new();
+    public ServersController(IServersRepository serverRep)
+    {
+        _serversRep = serverRep;
+    }
+
+    [HttpGet]
+    public ActionResult<List<ServerList>> GetAll()
+    {
+        return Ok(_serversRep.GetAll());
+    }
+
+    [HttpPost("/api/servers/add")]
+    public ActionResult<ServerList> Add(CustomServer server)
+    { 
+        ServerList? servers = _serversRep.GetName(server.Name);
+        ServerList? serversIp = _serversRep.GetIp(server.Ip);
+        Regex ip = new Regex(@"(([01]?\d\d?|2[0-4]\d|25[0-5])\.){3}([01]?\d\d?|2[0-4]\d|25[0-5])");
+        if (!ip.IsMatch(server.Ip))
+        {
+            return NotFound("Ip is incorrect !");
+        }
```

Comment:
Well done! 👍 But we can move this part to the beginning, so there would be less call to your data storage

Reply:
Also, it's might be better to change `NotFound` response code to `BadRequest`. Because we don't actually care if sever found or not, we just want to notify client that IP, inside request, is incorrect. So, basically, request is bad)